R/update-news.R

```diff
@@ -171,3 +207,122 @@ conventional_commit_types <- function() {
     "Testing" = "test"
   )
 }
+
+is_merge_commit <- function(message) {
+  grepl("^Merge pull request #([0-9]*) from", message)
+}
+
+harvest_pr_data <- function(message) {
+  check_gh_pat()
+
+  pr_number <- regmatches(message, regexpr("#[0-9]*", message))
+  pr_number <- sub("#", "", pr_number)
+
+  slug <- github_slug()
+
+  failure_message <- sprintf("Could not get title for PR #%s", pr_number)
+
+  pr_info <- if (!has_internet()) {
+    cli::cli_alert_warning(sprintf("%s (no internet connection)", failure_message))
+    NULL
+  } else {
+    tryCatch(
+      {
+        gh::gh(glue("GET /repos/{slug}/pulls/{pr_number}"))
+      },
+      error = function(e) {
+        print(e)
+        cli::cli_alert_warning(failure_message)
+        return(NULL)
+      }
+    )
+  }
+
+  tibble::tibble(
+    title = pr_info$title %||% NA,
```

Comment:
Minor nit -- for type stability:

```suggestion
    title = pr_info$title %||% NA_character_,
```